src/dc_receive_imf.rs

```diff
@@ -1414,6 +1444,79 @@ async fn create_or_lookup_group(
     Ok((chat_id, chat_id_blocked))
 }
 
+#[allow(clippy::indexing_slicing)]
+async fn create_or_lookup_mailinglist(
+    context: &Context,
+    allow_creation: bool,
+    create_blocked: Blocked,
+    list_id_header: &str,
+    subject: &str,
+) -> (ChatId, Blocked) {
+    static LIST_ID: Lazy<Regex> = Lazy::new(|| Regex::new(r"^(.*.)<(.*.)>$").unwrap());
+    let (mut name, listid) = match LIST_ID.captures(list_id_header) {
+        Some(cap) => (cap[1].trim().to_string(), cap[2].trim().to_string()),
+        None => (
+            "".to_string(),
+            list_id_header
+                .trim()
+                .trim_start_matches('<')
+                .trim_end_matches('>')
+                .to_string(),
+        ),
+    };
+
+    if let Ok((chat_id, _, blocked)) = chat::get_chat_id_by_grpid(context, &listid).await {
+        return (chat_id, blocked);
+    }
+
+    static SUBJECT: Lazy<Regex> = Lazy::new(|| Regex::new(r"^.{0,5}\[(.*.)\]").unwrap());
+    if let Some(cap) = SUBJECT.captures(subject) {
+        name = cap[1].to_string();
+    }
+
+    if name.is_empty() {
+        name = context
+            .stock_str(StockMessage::UnnamedNewsletter)
```

Comment:
ping @hpk42 as you also were involved in this discussion

We discussed exactly this back then in BB and we decided to change it from the listid to "Unnamed newsletter", I don't know why anymore, probably because the list id is often/sometimes cryptic. This doesn't mean I'm against changing this a second time, just so we don't change it back and forth all the time :)

Also, thinking about it - Whatever we decide here, sometimes the name we choose is going to be unfitting because we don't get a fitting name. For this case I think we should allow the user to change the name of the mailing list. We also allow users to change the name of their contacts and they don't expect that this name change is sent to anyone else.

Reply:
so, let's go for that, seems more straigt-forward to me.  it we see that that is not helpful, we can go for "unnamed sth." later on. i'll do a commit.